### ISSUE TITLE
Doesn't work on fresh windows because of VC Runtime external dependencies

### ISSUE THREAD

**Issue by splix** [ethereumproject/emerald-rs#173]
![emerald-windows](https://user-images.githubusercontent.com/48364/27715615-68846dea-5d06-11e7-9352-bcd26ef158aa.png)

and it's unclear where to get that DLL. do we need to ask users to install it before using emerald?

**Comment by dulanov** [comment]
Hh, Visual Studio Runtime, I think it is needed only to compile, I will try to find why.

**Comment by splix** [comment]
It seems that it works on other users machines. That one was completely fresh, only Chrome browser was installed before Emerald.

**Comment by dulanov** [comment]
Did you build the release version with `cargo build --release`?

**Comment by splix** [comment]
It was installed with `cargo install emerald-cli -f`

**Comment by dulanov** [comment]
```
> ldd  /c/Users/dulanov/.cargo/bin/emerald.exe

        ntdll.dll => /c/Windows/SYSTEM32/ntdll.dll (0x77a10000)
        kernel32.dll => /c/Windows/system32/kernel32.dll (0x777f0000)
        KERNELBASE.dll => /c/Windows/system32/KERNELBASE.dll (0x7fefd7c0000)
        ADVAPI32.dll => /c/Windows/system32/ADVAPI32.dll (0x7fefdd40000)
        msvcrt.dll => /c/Windows/system32/msvcrt.dll (0x7fefecf0000)
        sechost.dll => /c/Windows/SYSTEM32/sechost.dll (0x7feffd00000)
        RPCRT4.dll => /c/Windows/system32/RPCRT4.dll (0x7fefe670000)
        WS2_32.dll => /c/Windows/system32/WS2_32.dll (0x7fefec80000)
        NSI.dll => /c/Windows/system32/NSI.dll (0x7fefec70000)
        SHELL32.dll => /c/Windows/system32/SHELL32.dll (0x7fefed90000)
        SHLWAPI.dll => /c/Windows/system32/SHLWAPI.dll (0x7fefdac0000)
        GDI32.dll => /c/Windows/system32/GDI32.dll (0x7fefdfb0000)
        USER32.dll => /c/Windows/system32/USER32.dll (0x77910000)
        LPK.dll => /c/Windows/system32/LPK.dll (0x7fefe830000)
        USP10.dll => /c/Windows/system32/USP10.dll (0x7fefe3f0000)
        VCRUNTIME140.dll => /c/Windows/system32/VCRUNTIME140.dll (0x7fef7ce0000)
        api-ms-win-crt-runtime-l1-1-0.dll => /c/Windows/system32/api-ms-win-crt-runtime-l1-1-0.dll (0x7fefbb80000)
        ucrtbase.DLL => /c/Windows/system32/ucrtbase.DLL (0x7fee9ea0000)
        api-ms-win-core-timezone-l1-1-0.dll => /c/Windows/system32/api-ms-win-core-timezone-l1-1-0.dll (0x7fefafc0000)
        api-ms-win-core-file-l2-1-0.dll => /c/Windows/system32/api-ms-win-core-file-l2-1-0.dll (0x7fef5d00000)
        api-ms-win-core-localization-l1-2-0.dll => /c/Windows/system32/api-ms-win-core-localization-l1-2-0.dll (0x7fef5cf0000)
        api-ms-win-core-synch-l1-2-0.dll => /c/Windows/system32/api-ms-win-core-synch-l1-2-0.dll (0x7fef6a90000)
        api-ms-win-core-processthreads-l1-1-1.dll => /c/Windows/system32/api-ms-win-core-processthreads-l1-1-1.dll (0x7fef5ce0000)
        api-ms-win-core-file-l1-2-0.dll => /c/Windows/system32/api-ms-win-core-file-l1-2-0.dll (0x7fef5cd0000)
        api-ms-win-crt-string-l1-1-0.dll => /c/Windows/system32/api-ms-win-crt-string-l1-1-0.dll (0x7fef5cc0000)
        api-ms-win-crt-heap-l1-1-0.dll => /c/Windows/system32/api-ms-win-crt-heap-l1-1-0.dll (0x7fef5cb0000)
        api-ms-win-crt-stdio-l1-1-0.dll => /c/Windows/system32/api-ms-win-crt-stdio-l1-1-0.dll (0x7fef5ca0000)
        api-ms-win-crt-convert-l1-1-0.dll => /c/Windows/system32/api-ms-win-crt-convert-l1-1-0.dll (0x7fef5c90000)
        api-ms-win-crt-math-l1-1-0.dll => /c/Windows/system32/api-ms-win-crt-math-l1-1-0.dll (0x7fef5c80000)
        api-ms-win-crt-locale-l1-1-0.dll => /c/Windows/system32/api-ms-win-crt-locale-l1-1-0.dll (0x7fef5c70000)
```

**Comment by dulanov** [comment]
```
> pldd 4824
4824:   /c/Users/dulanov/.cargo/bin/emerald
/c/Windows/SYSTEM32/ntdll.dll
/c/Windows/system32/kernel32.dll
/c/Windows/system32/KERNELBASE.dll
/c/Windows/system32/ADVAPI32.dll
/c/Windows/system32/msvcrt.dll
/c/Windows/SYSTEM32/sechost.dll
/c/Windows/system32/RPCRT4.dll
/c/Windows/system32/WS2_32.dll
/c/Windows/system32/NSI.dll
/c/Windows/system32/SHELL32.dll
/c/Windows/system32/SHLWAPI.dll
/c/Windows/system32/GDI32.dll
/c/Windows/system32/USER32.dll
/c/Windows/system32/LPK.dll
/c/Windows/system32/USP10.dll
/c/Windows/system32/VCRUNTIME140.dll
/c/Windows/system32/api-ms-win-crt-runtime-l1-1-0.dll
/c/Windows/system32/ucrtbase.DLL
/c/Windows/system32/api-ms-win-core-timezone-l1-1-0.dll
/c/Windows/system32/api-ms-win-core-file-l2-1-0.dll
/c/Windows/system32/api-ms-win-core-localization-l1-2-0.dll
/c/Windows/system32/api-ms-win-core-synch-l1-2-0.dll
/c/Windows/system32/api-ms-win-core-processthreads-l1-1-1.dll
/c/Windows/system32/api-ms-win-core-file-l1-2-0.dll
/c/Windows/system32/api-ms-win-crt-string-l1-1-0.dll
/c/Windows/system32/api-ms-win-crt-heap-l1-1-0.dll
/c/Windows/system32/api-ms-win-crt-stdio-l1-1-0.dll
/c/Windows/system32/api-ms-win-crt-convert-l1-1-0.dll
/c/Windows/system32/api-ms-win-crt-math-l1-1-0.dll
/c/Windows/system32/api-ms-win-crt-locale-l1-1-0.dll
/c/Windows/system32/IMM32.DLL
/c/Windows/system32/MSCTF.dll
/c/Windows/system32/CRYPTSP.dll
/c/Windows/system32/rsaenh.dll
/c/Windows/system32/CRYPTBASE.dll
/c/Windows/system32/mswsock.dll
/c/Windows/System32/wshtcpip.dll
```

**Comment by dulanov** [comment]
More details can be found here - https://www.reddit.com/r/rust/comments/61pjbp/static_linking_support/

**Comment by dulanov** [comment]
For now the only recommendation - install `Visual C++ Redistributable for Visual Studio 2015` - https://www.microsoft.com/en-us/download/details.aspx?id=48145, or rebuild with GNU toolchain, but it doen't work now because of `rust-crypto` crate.

**Comment by dulanov** [comment]
I was able to reproduce the error locally 
![image](https://user-images.githubusercontent.com/39793/28119213-4df66948-6714-11e7-8c9b-9146a61c828c.png)

**Comment by dulanov** [comment]
@ia can you check it - https://github.com/electron/electron/pull/5538, that problem should be fixed in Electron v1.1.1

**Comment by dulanov** [comment]
https://github.com/electron/electron/pull/5538/commits/8ed3659bb85685097552c584a8c20660079e916c

**Comment by whilei** [comment]
I'm not sure I understand exactly the question -- wallet is using `"electron": "^1.6.11",` currently -- 

I wonder if the following might help? It should compile Electron app dependencies for existing system.
```
$ npm run postinstall
```
> `"postinstall": "electron-builder install-app-deps",`

**Comment by dulanov** [comment]
@ia electron contains Visual C++ runtimes, i try to find why we don't have them yet, because:

> Electron version: 1.1.0
Operating system: Windows
This release contains the Visual C++ 2013 runtimes instead of Visual C++ 2015 runtimes.

**Comment by dulanov** [comment]
Probably exists some command to update external binaries? From v1.1.1 electron start to use exactly VC++ 2015 runtime, it should work!

**Comment by dulanov** [comment]
https://github.com/electron/electron/blob/master/script/update-external-binaries.py

**Comment by dulanov** [comment]
What the hell? And then delete it in https://github.com/electron/electron/commit/4702d8b640a3ed061f1785b5b579cc8b15c97ee9#diff-81bd91b359c3a9857fba4a7f0f448d8e

**Comment by dulanov** [comment]
![image](https://user-images.githubusercontent.com/39793/28124067-ee904a5a-6722-11e7-9f78-74db01f73c34.png)

**Comment by dulanov** [comment]
Ok, it should be preinstalled by user - https://github.com/electron/electron/blob/master/script/create-dist.py#L138

**Comment by dulanov** [comment]
![image](https://user-images.githubusercontent.com/39793/28125064-c73e9fbc-6725-11e7-8a28-2eb41eae0472.png)

**Comment by dulanov** [comment]
Is fixed by #193.
